src/core/mac/mac.cpp

```diff
@@ -2269,6 +2276,7 @@ void Mac::SetCslChannel(uint8_t aChannel)
     VerifyOrExit(GetCslChannel() != aChannel, OT_NOOP);
 
     mSubMac.SetCslChannel(aChannel);
+    mSubMac.SetCslChannelSpecified(true);
```

Comment:
is there chance that the user may want to clear the csl channel he/she specified and let csl work on the same pan channel?

Reply:
Make sense. As I mentioned above, I think users can do this by setting CSL channel to 0 again.